frontend/src/Commons/Forms/InputModules/Crops/CropsInfo.tsx

```diff
@@ -0,0 +1,272 @@
+import CustomField from '@Commons/Input/Field/CustomField';
+import CropsButtonGroup from './CropsButtonGroup';
+import {
+  StyledCropsSmallGroup,
+  StyledCropsLargeGroup,
+  StyledAddCancelButtonGroup,
+} from './CropsInfo.styles';
+import {
+  StyledFarmInfo,
+  StyledButtonGroupContainer,
+  StyledAreaContainer,
+} from '../../../FormStyles.styles';
+
+const CropsInfoComponent: React.FC<InputModuleProps> = ({
+  farmDetails,
+  updateFarmDetails,
+  handleFormState,
+}) => {
+  const [, setFieldsInfo] = useState(farmDetails);
+  const [fieldIndex, setFieldIndex] = useState(0);
+  const [cropIndex, setCropIndex] = useState(0);
+  const [initialFieldValues, setInitialFieldValues] = useState(CropsInitialDetails);
+  // Only triggered once, it would show list and persists.
+  const [isSubmitted, setSubmitted] = useState<boolean>(false);
+  // Would trigger when new field button is clicked.
+  const [isButtonDisplayed, setButtonDisplayed] = useState<boolean>(false);
+  const validationSchema = Yup.object().shape({
+    cropId: Yup.string().required('Required'),
+    yield: Yup.number().positive().max(100).required('Required'),
+    plantAgeYears: Yup.string().when('cropId', (cropId) => (cropId.toString() === 'Blueberry'
+      ? Yup.string().required('Required')
+      : Yup.string().notRequired())),
+    numberOfPlantsPerAcre: Yup.number().when('cropId', (cropId) => (cropId.toString() === 'Blueberry'
+      ? Yup.number().required('Required')
+      : Yup.number().notRequired())),
+    distanceBtwnPlants: Yup.string().when('cropId', (cropId) => (cropId.toString() === 'Blueberry'
+      ? Yup.string().required('Required')
+      : Yup.string().notRequired())),
+    distanceBtwnRows: Yup.string().when('cropId', (cropId) => (cropId.toString() === 'Blueberry'
+      ? Yup.string().required('Required')
+      : Yup.string().notRequired())),
+    willPlantsBePruned: Yup.boolean().required('Required'),
+    whereWillPruningsGo: Yup.string().required('Required'),
+    willSawdustBeApplied: Yup.boolean().required('Required'),
+  });
+
+  const distanceBtwnPlantsRowsDataAppend = (values: SubmissionCropsInterface) => {
+    let resultString = '';
+
+    if (values.distanceBtwnPlants === '0.6' && values.distanceBtwnRows === '2.7') {
+      resultString += '(2ft x 9ft)';
+    }
+    if (values.distanceBtwnPlants === '0.6' && values.distanceBtwnRows === '3.0') {
+      resultString += '(2ft x 10ft)';
+    }
+    if (values.distanceBtwnPlants === '0.75' && values.distanceBtwnRows === '2.7') {
+      resultString += '(2.5ft x 9ft)';
+    }
+    if (values.distanceBtwnPlants === '0.75' && values.distanceBtwnRows === '3.0') {
+      resultString += '(2.5ft x 10ft)';
+    }
+    if (values.distanceBtwnPlants === '0.9' && values.distanceBtwnRows === '2.7') {
+      resultString += '(3ft x 9ft)';
+    }
+    if (values.distanceBtwnPlants === '0.9' && values.distanceBtwnRows === '3.0') {
+      resultString += '(3ft x 10ft)';
+    }
+
+    return resultString;
+  };
```

Comment:
Great candidate for a switch if you want to optimize this a bit.

Reply:
Makes sense alright I'll do switch